usecases/webhook_events_usecase.go

```diff
@@ -112,9 +113,9 @@ func (usecase WebhookEventsUsecase) SendWebhookEventAsync(ctx context.Context, w
 	logger := utils.LoggerFromContext(ctx).With("webhook_event_id", webhookEventId)
 	ctx = utils.StoreLoggerInContext(ctx, logger)
 
-	// go routine to send the webhook event asynchronously, with a new context and timeout and a child span
+	// goroutine to send the webhook event asynchronously, with a new context and timeout and a child span
 	go func() {
-		ctx, cancel := context.WithTimeout(context.WithoutCancel(ctx), time.Second*3)
+		ctx, cancel := context.WithTimeout(context.WithoutCancel(ctx), ASYNC_WEBHOOKS_SEND_TIMEOUT)
```

Comment:
A remark about this (I struggled a bit with this and unexpected errors at first).
The rate limiter will return an error if we call Wait() and the expected wait time (considering other requests on the rate limiter instance) is longer than the remaining context duration, considering the configured refresh rate.
With a 5 minute timeout and a 25/sec refresh rate (corresponding to a 50/res rate limit on convoy configured in env variable), that's 25*60*5=7500 events max that can be queued before the internal rate limiter starts dropping some (the initial commit would have been only 25*3=75 events, so that batch retry jobs would still not quite have worked).